client/src/scenes/Home/SyncRequestForm/components/ColorPicker.tsx

```diff
@@ -20,59 +21,67 @@ const useStyles = makeStyles(() =>
             height: '20px',
             borderRadius: 2,
             paddingLeft: '10px',
+            paddingRight: '10px',
             backgroundClip: 'content-box',
         },
     }),
 );
 
 interface Color {
-    id: number;
-    name: string;
-    color: string;
+    [key: string]: {
+        name: string;
+        color: string;
+    };
 }
 
 export const initialColorId = 0;
 
-export const ColorPicker = ({ onChange = noop }: ColorPickerProps) => {
+export const ColorPicker = ({ onChange = noop, defaultColor }: ColorPickerProps) => {
     const style = useStyles();
-    const colors: Color[] = [
-        { id: 0, name: 'Default', color: '#ffffff' },
-        { id: 1, name: 'Lavender', color: '#7986cb' },
-        { id: 2, name: 'Sage', color: '#33b679' },
-        { id: 3, name: 'Grape', color: '#8e24aa' },
-        { id: 4, name: 'Flamingo', color: '#e67c73' },
-        { id: 5, name: 'Banana', color: '#f6c026' },
-        { id: 6, name: 'Tangerine', color: '#f5511d' },
-        { id: 7, name: 'Peacock', color: '#039be5' },
-        { id: 8, name: 'Graphite', color: '#616161' },
-        { id: 9, name: 'Blueberry', color: '#3f51b5' },
-        { id: 10, name: 'Basil', color: '#0b8043' },
-        { id: 11, name: 'Tomato', color: '#d60000' },
-    ];
+    const colors: Color = {
+        '1': { name: 'Lavender', color: '#7986cb' },
+        '2': { name: 'Sage', color: '#33b679' },
+        '3': { name: 'Grape', color: '#8e24aa' },
+        '4': { name: 'Flamingo', color: '#e67c73' },
+        '5': { name: 'Banana', color: '#f6c026' },
+        '6': { name: 'Tangerine', color: '#f5511d' },
+        '7': { name: 'Peacock', color: '#039be5' },
+        '8': { name: 'Graphite', color: '#616161' },
+        '9': { name: 'Blueberry', color: '#3f51b5' },
+        '10': { name: 'Basil', color: '#0b8043' },
+        '11': { name: 'Tomato', color: '#d60000' },
+
+        get '0'() {
+            if (defaultColor === undefined) {
+                return { name: 'Default', color: '#ffffff' };
+            }
+            return { name: 'Default', color: defaultColor };
+        },
+    };
 
     return (
         <div>
-            <FormControl className={style.formControl}>
+            <FormControl className={style.formControl} disabled={defaultColor === undefined}>
```

Comment:
We should definitely suppress the default option if there is none instead of disabling the form control